src/client.rs

```diff
@@ -164,6 +164,199 @@ impl Client {
+    /// the WebDriver client will navigate to in order to retrieve the cookies needed for 
+    /// the raw HTTP request.
+    ///
+    /// This approach is necessary due to the WebDriver limitation discussed in 
+    /// [w3c/webdriver#1238](https://github.com/w3c/webdriver/issues/1238), 
+    /// which prevents setting cookies for a domain that the WebDriver is not currently on.
+    /// 
+    /// By setting this URL, you can ensure that the appropriate cookies are included in the 
+    /// raw HTTP request. This can be particularly useful for scenarios where you need to 
+    /// reuse cookies from a previous session to avoid redundant login operations or share 
+    /// WebDriver sessions across different threads with distinct cookies.
+    ///
+    /// - [Issue #148](https://github.com/jonhoo/fantoccini/issues/148)
+    pub fn cookie_url(&mut self, url: &str) -> &mut Self {
+        self.cookie_url = Some(url.to_string());
+        self
+    }
+
+    /// Set a function to modify the request.
+    pub fn map_request<F>(&mut self, f: F) -> &mut Self
+    where
+        F: FnOnce(http::request::Builder) -> hyper::Request<BoxBody<hyper::body::Bytes, Infallible>> + Send + 'static,
+    {
+        self.request_modifier = Some(Box::new(f));
+        self
+    }
+
+    /// Send the constructed request.
+    pub async fn send(self) -> Result<hyper::Response<hyper::body::Incoming>, error::CmdError> {
+        let url = self.url;
+        let method = self.method;
+        let cookie_url = self.cookie_url;
+        let request_modifier = self.request_modifier;
+
+        // We need to do some trickiness here. GetCookies will only give us the cookies for the
+        // *current* domain, whereas we want the cookies for `url`'s domain. So, we navigate to the
+        // URL in question, fetch its cookies, and then navigate back. *Except* that we can't do
+        // that either (what if `url` is some huge file?). So we *actually* navigate to some weird
+        // url that's unlikely to exist on the target domain, and which won't resolve into the
+        // actual content, but will still give the same cookies.
+        //
+        // The fact that cookies can have /path and security constraints makes this even more of a
+        // pain. /path in particular is tricky, because you could have a URL like:
+        //
+        //    example.com/download/some_identifier/ignored_filename_just_for_show
+        //
+        // Imagine if a cookie is set with path=/download/some_identifier. How do we get that
+        // cookie without triggering a request for the (large) file? I don't know. Hence: TODO.
+        //
+        // Retrieve cookies and User-Agent from the WebDriver session if a cookie URL is provided.
+        let (cookies, ua) = if let Some(cookie_url) = cookie_url {
+            // Navigate to the cookie URL to retrieve cookies.
+            self.client.goto(&cookie_url).await?;
```

Comment:
The old code would `.join` the current URL with the cookie url — I feel like we may want to do that here as well so that users can provide a relative URL to `.cookie_url` instead of an absolute one?

Reply:
Yes, I've updated the code to `.join` the current URL with the `cookie url`.